src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -328,21 +328,21 @@ private IndexedNode(int bitmap, List<AbstractNode<K, V>> subNodes) {
             int bit = mask;
             int count = 0;
             List<AbstractNode<K, V>> sub = subNodes;
-            List<AbstractNode<K, V>> arr = List.empty();
+            final Object[] arr = new Object[BUCKET_SIZE];
```

Comment:
Also in Array. Wouldn't it be better to make back of type `T[]` and additionally provide factory methods to create typed, empty arrays of a specific size? E.g. `static <T> T[] create(int size)` or maybe `static <T> T[] create(int size, Class<T> type)`. Then the package internal `wrap` method can be possibly removed.

```java
// T[] instead?
private final Object[] back;

private Array(Object[] back) {
    this.back = back;
}
```